Do not create and resolve directories for resolved directories in WebsiteDataStoreCocoa.mm

szewai opened the pull request:
#### 12cc91772f497aab979e90bbcfa6bdfc0a3037d1
<pre>
Do not create and resolve directories for resolved directories in WebsiteDataStoreCocoa.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=242908">https://bugs.webkit.org/show_bug.cgi?id=242908</a>

Reviewed by Geoffrey Garen.

To reduce unnecessary syscalls.

* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::platformSetNetworkParameters):
(WebKit::WebsiteDataStore::resolvedContainerTemporaryDirectory):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):
(WebKit::WebsiteDataStore::createHandleFromResolvedPathIfPossible):
(WebKit::createHandleFromResolvedPathIfPossible): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:

Canonical link: <a href="https://commits.webkit.org/252640@main">https://commits.webkit.org/252640@main</a>
</pre>

geoffreygaren commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
```diff
@@ -317,8 +317,8 @@ void WebsiteDataStore::resolveDirectoriesIfNecessary()
     if (!m_configuration->cookieStorageFile().isEmpty()) {
-        m_resolvedConfiguration->setCookieStorageFile(resolveAndCreateReadWriteDirectoryForSandboxExtension(FileSystem::parentPath(m_configuration->cookieStorageFile())));
-        m_resolvedConfiguration->setCookieStorageFile(FileSystem::pathByAppendingComponent(m_resolvedConfiguration->cookieStorageFile(), FileSystem::pathFileName(m_configuration->cookieStorageFile())));
+        auto resovledCookieDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(FileSystem::parentPath(m_configuration->cookieStorageFile()));
```
resolvedCookieDirectory

geoffreygaren reviewed (approved):
r=me

szewai committed:
Do not create and resolve directories for resolved directories in WebsiteDataStoreCocoa.mm
https://bugs.webkit.org/show_bug.cgi?id=242908

Reviewed by Geoffrey Garen.

To reduce unnecessary syscalls.

* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::platformSetNetworkParameters):
(WebKit::WebsiteDataStore::resolvedContainerTemporaryDirectory):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):
(WebKit::WebsiteDataStore::createHandleFromResolvedPathIfPossible):
(WebKit::createHandleFromResolvedPathIfPossible): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:

Canonical link: https://commits.webkit.org/252640@main
